core/src/main/java/org/openedx/core/presentation/dialog/IAPDialogFragment.kt

```diff
@@ -0,0 +1,319 @@
+                                                iapException.getFormattedErrorMessage()
+                                            )
+                                            onDismiss()
+                                        },
+                                        onDismiss = {
+                                            iapViewModel.logIAPErrorActionEvent(
+                                                iapException.requestType.request,
+                                                IAPAction.ACTION_CLOSE.action
+                                            )
+                                            onDismiss()
+                                        })
+                                }
+
+                                (iapException.requestType == IAPRequestType.EXECUTE_ORDER_CODE) -> {
+                                    UpgradeErrorDialog(
+                                        title = stringResource(id = R.string.iap_error_title),
+                                        description = stringResource(id = R.string.iap_course_not_fullfilled),
+                                        confirmText = stringResource(id = R.string.core_error_try_again),
+                                        onConfirm = {
+                                            iapViewModel.logIAPErrorActionEvent(
+                                                iapException.requestType.request,
+                                                IAPAction.ACTION_RETRY.action
+                                            )
+                                            iapViewModel.retryExecuteOrder()
+                                        },
+                                        dismissText = stringResource(id = R.string.core_cancel),
+                                        onDismiss = {
+                                            iapViewModel.logIAPErrorActionEvent(
+                                                iapException.requestType.request,
+                                                IAPAction.ACTION_CLOSE.action
+                                            )
+                                            onDismiss()
+                                        }
+                                    )
+                                }
+
+                                else -> {
+                                    val description: String = when {
+                                        (iapException.httpErrorCode == 403) -> {
+                                            stringResource(id = R.string.iap_unauthenticated_account_message)
+                                        }
+
+                                        (iapException.httpErrorCode == 400) -> {
+                                            stringResource(id = R.string.iap_course_not_available_message)
+                                        }
+
+                                        else -> {
+                                            ""
```

Comment:
We had a default error message as well.

Reply:
no need cuz we are adding default one inside the compose method if parameter is null